Automattic/wp-calypso
# Fix/desktop loader

This PR allows the desktop server to boot up after the changes in https://github.com/Automattic/wp-calypso/pull/30768 and https://github.com/Automattic/wp-calypso/pull/32253. 

#### Changes proposed in this Pull Request
Fix how we detect the desktop environment, also checking for desktop-development, and fix the page we serve to use the fallback path for production builds.

#### Testing instructions
1. Check out the `release/4.2.0` branch in WP-Desktop. The branch is already set up to reference the commits in this PR. Also, there are some related changes in that branch that are not in `develop` yet.
2. Run the `make dev-server` to start the development server.
3. In a separate terminal run `make dev`.
4. Verify that Calypso comes up inside the desktop shell.
5. Exit the desktop app and kill the dev server.
6. Run `make build` to build the production app.
7. Run the artifacts in the `release` folder  and verify that Calypso comes up inside the desktop shell without errors.

See also the relevant discussion in https://github.com/Automattic/wp-calypso/pull/32738.
